SofaKernel/framework/sofa/core/objectmodel/Base.h

```diff
@@ -332,17 +334,13 @@ class SOFA_CORE_API Base
 
     /// @}
 
-    /// @name logs
-    ///   Messages and warnings logging
-    /// @{
 
 private:
-
     /// effective ostringstream for logging
     mutable std::ostringstream _serr, _sout;
+    mutable std::deque<sofa::helper::logging::Message> m_messageslog ;
```

Comment:
Actually I was thinking you asked to implement this feature otherwise the msg_* api wouln't be equivalent enough to the sout/serr one. 

I personally have no clear idea whether or not we could in some future remove the per component log. 

So this PR is more about improving the implementation of the existing approach and improving the way sofa handle message. 

Gven the way msg_* is designed removing the per object backlog and replace it with a single history wouldn't be a hard if someone is willing to do that.